alert/alertlog/store.go

```diff
@@ -316,46 +310,34 @@ func (s *Store) logEntry(ctx context.Context, tx *sql.Tx, _type Type, meta inter
 				r.subject.classifier = "no immediate rule"
 				break
 			}
-			cmType, err := s.queries(tx).AlertLogLookupCMType(ctx, uuid.MustParse(src.ID))
+			dest, err := s.queries(tx).AlertLogLookupCMDest(ctx, uuid.MustParse(src.ID))
 			if err != nil {
 				return nil, errors.Wrap(err, "lookup contact method type for callback ID")
 			}
-			switch cmType {
-			case gadb.EnumUserContactMethodTypeVOICE:
-				r.subject.classifier = "Voice"
-			case gadb.EnumUserContactMethodTypeSMS:
-				r.subject.classifier = "SMS"
-			case gadb.EnumUserContactMethodTypeEMAIL:
-				r.subject.classifier = "Email"
-			case gadb.EnumUserContactMethodTypeWEBHOOK:
-				r.subject.classifier = "Webhook"
-			case gadb.EnumUserContactMethodTypeSLACKDM:
-				r.subject.classifier = "Slack"
+			info, err := s.reg.TypeInfo(ctx, dest.DestV1.Type)
+			if err != nil {
+				return nil, errors.Wrap(err, "lookup contact method type")
 			}
+			r.subject.classifier = info.Name
 
 		case permission.SourceTypeNotificationCallback:
 			r.subject._type = SubjectTypeUser
-			var dt notification.ScannableDestType
-			err = txWrap(ctx, tx, s.lookupCallbackType).QueryRowContext(ctx, src.ID).Scan(&dt.CM, &dt.NC)
+			r.subject.userID = permission.UserNullUUID(ctx)
+
+			id, err := uuid.Parse(src.ID)
 			if err != nil {
-				return nil, errors.Wrap(err, "lookup notification type for callback ID")
+				return nil, errors.Wrap(err, "parse channel ID")
```

Comment:
is this always a channel (thinking as in slack channel)?
```suggestion
				return nil, errors.Wrap(err, "parse destination source ID")
```

Reply:
NVM realizing this is probably meaning notification channel

Reply:
yes, regardless of type it's the UUID of a row in the `notification_channels` table